### PR TITLE
Ran pyupgrade to upgrade string formatting syntax.

### PR DESCRIPTION
**Pull request by adamchainz**
pyupgrade does a number of rewrites but after #14897 and #14902 the only remaining ones for Django are the string rewrites.

printf style formatting is upgraded to `str.format()`: https://github.com/asottile/pyupgrade/#printf-style-string-formatting
`str.format()` is then upgraded to an f-string, for simple cases only: https://github.com/asottile/pyupgrade/#f-strings

> `pyupgrade` is intentionally timid and will not create an f-string if it would make the expression longer or if the substitution parameters are sufficiently complicated (as this can decrease readability).

I've looked through the results and I think they'd all pass Django's policy on f strings.

### CODE REVIEW

**Comment by felixxm** [comment]
@adamchainz Thanks for this patch :+1: , however we agreed on [the mailing list](https://groups.google.com/g/django-developers/c/XNhpOHam0zE/m/Nw7hfEoaBgAJ) that any bulk updates should be avoided. We also don't prefer any particular formatting:

> "String variable interpolation may use %-formatting, f-strings, or str.format() as appropriate, with the goal of maximizing code readability."

**Comment by adamchainz** [comment]
Thanks, I forgot to look up that thread. I guess there is a consensus against it then.

That's a shame, pyupgrade does a bunch of neat things that would save time in code review.h, but to do so it ofc has to be opinonated.

**Comment by ngnpope** [comment]
We are where we are at the moment regarding the decision not to reformat...

...but when we introduce `black`, which will reformat string quoting, etc., I think that we could also consider `pyupgrade`.

It seems a shame to not be able to use such a useful tool to prevent old rot seeping back in. Generally I've always preferred single quotes, but I see the appeal of `black` to help save time. Equally, I'm not much of a fan of `.format()`. It's more verbose and slower than %-formatting, but I think f-strings are great and `pyupgrade` makes this easy. We'd probably want to manually convert those `.format()` calls that are not converted to f-strings where appropriate.

If both tools were run sequentially we could then add the few commit hashes to the ignored revisions file we were planning to create when we start using `black`.

Obviously this is my view and we'd need to take it to the mailing list again.

TL;DR: If we're going to be reformatting everything with black, I don't see why we cannot also fix up to use f-strings more widely and, as a result, be able to add `pyupgrade` to our linting arsenal. In light of the former, I don't see the latter as being any worse/different.

**Comment by felixxm** [comment]
> TL;DR: If we're going to be reformatting everything with black, I don't see why we cannot also fix up to use f-strings more widely and, as a result, be able to add `pyupgrade` to our linting arsenal. In light of the former, I don't see the latter as being any worse/different.

The question is how many linters we will use? `flake`, `isort`, `black`, `pyupgrade`, `pylint` etc. I really don't see any benefits of changing `%`-formatting to the `.format()`. We drop support for Python versions once per two years I think it is not worth using `pyupgrade` on daily basis :shrug: . My 2¢.

**Comment by ngnpope** [comment]
> We drop support for Python versions once per two years I think it is not worth using `pyupgrade` on daily basis 🤷 . My 2¢.

I think the "upgrade" in the project name makes it confusing. It is possible to use it to prevent new use of old patterns that have previously been "upgraded" using the tool.

**Comment by ngnpope** [comment]
@adamchainz We could use `--keep-percent-format` for now?

**Comment by felixxm** [comment]
> I think the "upgrade" in the project name makes it confusing. It is possible to use it to prevent new use of old patterns that have previously been "upgraded" using the tool.

Without changes in the string formatting it suggested two changes related with `subprocess.run()` and `@lru_cache` ~30 LOC. IMO calling it once before each feature freeze should be enough.

**Comment by carltongibson** [comment]
Thanks all. 

TBH I don't think adding another linter here now is worth the price of entry, particularly thinking about new contributors, but also just trees 🌳

The consensus when allowing f-string usage was very much that we were not removing %-fomatting, and avoiding bulk edits for stylistic changes has been mentioned (to avoid) many times.  

Any of that is (as ever) open to reevaluation, but I'll close this here on the basis of the status quo.

**Comment by adamchainz** [comment]
Thanks for the discussion all.

> IMO calling it once before each feature freeze should be enough.

This is a fair call, if you want to add it to the appropriate checklist.
